src/inference/src/dev/threading/cpu_streams_executor.cpp

```diff
@@ -105,14 +125,25 @@ struct CPUStreamsExecutor::Impl {
                                    const int core_type,
                                    const int numa_node_id,
                                    const int socket_id,
-                                   const int max_threads_per_core) {
+                                   const int max_threads_per_core,
+                                   const bool pin_processors) {
             auto stream_processors = _impl->_config.get_stream_processor_ids();
             _numaNodeId = numa_node_id;
             _socketId = socket_id;
             if (stream_type == STREAM_WITHOUT_PARAM) {
                 _taskArena.reset(new custom::task_arena{custom::task_arena::constraints{}
                                                             .set_max_concurrency(concurrency)
                                                             .set_max_threads_per_core(max_threads_per_core)});
+                if (pin_processors) {
+                    CpuSet mask;
+                    int ncpus = 0;
+                    std::tie(mask, ncpus) = get_process_mask();
+                    CpuSet setMask = get_pecore_mask(ncpus);
```

Comment:
Please create mask based on stream information instead of pecore only.

Reply:
Updated.